src/WebJobs.Script/Extensions/IDictionaryExtensions.cs

```diff
@@ -19,5 +20,27 @@ public static class IDictionaryExtensions
                 dictionary[pair.Key] = pair.Value;
             }
         }
+
+        public static IDictionary<string, TValue> AsCaseInsensitive<TValue>(this IDictionary<string, TValue> dictionary)
+        {
+            return new Dictionary<string, TValue>(dictionary, StringComparer.OrdinalIgnoreCase);
```

Comment:
A dictionary cannot contain duplicates values. This conversion doesn't change that - it just changes the key comparer.